util/internal/integrationtest/backend/lucene/src/main/java/org/hibernate/search/util/impl/integrationtest/backend/lucene/directory/TrackingDirectoryProvider.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.util.impl.integrationtest.backend.lucene.directory;
+
+import org.hibernate.search.backend.lucene.lowlevel.directory.spi.DirectoryCreationContext;
+import org.hibernate.search.backend.lucene.lowlevel.directory.spi.DirectoryHolder;
+import org.hibernate.search.backend.lucene.lowlevel.directory.spi.DirectoryProvider;
+
+import org.apache.lucene.store.SingleInstanceLockFactory;
+
+
+public class TrackingDirectoryProvider implements DirectoryProvider {
```

Comment:
Actually the scope of this class is not clear to me. I was thinking if this class is supposed to be used by some user should be documented somehow. If not, maybe should be moved to the tests.

Reply:
It is in test code. More precisely, utils for integration tests:

```
package org.hibernate.search.util.impl.integrationtest.backend.lucene.directory
```

I'll add a short comment.